HCP Packer data sources causes Packer to crash on validate

<!--- Please keep this note for the community --->

#### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

#### Overview of the Issue

Running packer validate on an otherwise validate build template the contains an `hcp-packer-image` data source block causes Packer to crash. Running a build on the template does not result in the same error.

This issue does not exist in v1.8.0 - main culprit for this issue is likely https://github.com/hashicorp/packer/pull/11832 

#### Reproduction Steps

1. Run packer validate on template. 

### Packer version

v1.8.2

### Simplified Packer Template

```hcl
data "hcp-packer-iteration" "centos7-base" {
  bucket_name = var.hcp_bucket_name
  channel          = var.hcp_channel_name
}
data "hcp-packer-image" "centos7-base" {
  bucket_name    = data.hcp-packer-iteration.centos7-base.bucket_name
  iteration_id   = data.hcp-packer-iteration.centos7-base.id
  cloud_provider = "aws"
  region         = var.region
}
source "amazon-ebs" "centos7-web" {
  ami_name      = "${var.prefix}-centos7-web-${local.timestamp}"
  ami_regions   = ["${var.region}", "eu-west-1", "eu-north-1"]
  instance_type = "t2.micro"
  region        = var.region
  source_ami = data.hcp-packer-image.centos7-base.id
  ssh_username         = var.ssh_username
  ssh_private_key_file = var.ssh_private_key_file
  ssh_keypair_name     = var.ssh_keypair_name
}
build {
  hcp_packer_registry {
    bucket_name = "centos7-web-aws"
    description = "This is a CentOS7-based build that has been prepared as a web server in AWS."
    bucket_labels = {
      "build-version" = "0.0.1",
      "service"       = "aws"
    }
    build_labels = {
      "os"      = "centos7",
      "version" = "web"
    }
  }

  sources = ["source.amazon-ebs.centos7-web"]
}
```

### Operating system and Environment details
All
### Log Fragments and crash.log files

```shell
 packer validate .
panic: can't use ElementIterator on unknown value
​
goroutine 1 [running]:
github.com/zclconf/go-cty/cty.Value.ElementIterator({{{0x69f8348, 0xc000dd1860}}, {0x535f020, 0xa692de0}})
        /Users/runner/go/pkg/mod/github.com/zclconf/go-cty@v1.10.0/cty/value_ops.go:1118 +0xd8
github.com/zclconf/go-cty/cty.Value.AsValueMap({{{0x69f8348, 0xc000dd1860}}, {0x535f020, 0xa692de0}})
        /Users/runner/go/pkg/mod/github.com/zclconf/go-cty@v1.10.0/cty/value_ops.go:1324 +0xa5
github.com/hashicorp/packer/hcl2template.(*Parser).decodeBuildConfig(0x53bb660, 0xc000de91e0, 0xc00037b040)
        /Users/runner/work/packer/packer/hcl2template/types.build.go:267 +0x1b14
github.com/hashicorp/packer/hcl2template.(*Parser).parseConfig(0xc00037b040, 0xc0006ba160, 0xc00037b040)
        /Users/runner/work/packer/packer/hcl2template/parser.go:373 +0x1eb
github.com/hashicorp/packer/hcl2template.(*PackerConfig).Initialize(0xc00037b040, {0x0})
        /Users/runner/work/packer/packer/hcl2template/parser.go:326 +0x59a
github.com/hashicorp/packer/command.(*ValidateCommand).RunContext(0xc000810840, {0xc0001181a0, 0x1}, 0xc000a04000)
        /Users/runner/work/packer/packer/command/validate.go:59 +0x82
github.com/hashicorp/packer/command.(*ValidateCommand).Run(0xc000810840, {0xc0001181a0, 0x1, 0x1})
        /Users/runner/work/packer/packer/command/validate.go:25 +0xc5
github.com/mitchellh/cli.(*CLI).Run(0xc000152140)
        /Users/runner/go/pkg/mod/github.com/mitchellh/cli@v1.1.2/cli.go:262 +0x5f8
main.wrappedMain()
        /Users/runner/work/packer/packer/main.go:262 +0xaf2
main.realMain()
        /Users/runner/work/packer/packer/main.go:49 +0xf3
main.main()
        /Users/runner/work/packer/packer/main.go:35 +0x19
​
​
​
!!!!!!!!!!!!!!!!!!!!!!!!!!! PACKER CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
​
Packer crashed! This is always indicative of a bug within Packer.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Packer[1] so that we can fix this.
​
[1]: https://github.com/hashicorp/packer/issues
```

--- a/hcl2template/types.build.go
+++ b/hcl2template/types.build.go
@@ -264,6 +264,19 @@ func (p *Parser) decodeBuildConfig(block *hcl.Block, cfg *PackerConfig) (*BuildB
 		}
 
 		for _, value := range dsValues.AsValueMap() {
+			// Only try to check ancestry if we have fetched the
+			// data from HCP for the data source.
+			//
+			// NOTE: this will happen especially during validate as
+			// when we reach this point, we haven't fetched the
+			// data from HCP.
+			//
+			// TODO: maybe move this HCP-related logic outside that
+			// decode block so it's only executed during build?
+			if !value.IsKnown() {
+				continue
+			}
+
 			values := value.AsValueMap()
 			imgID, itID := values["id"], values["iteration_id"]
 			cfg.bucket.SourceImagesToParentIterations[imgID.AsString()] = itID.AsString()

